django/db/backends/sqlite3/creation.py

```diff
@@ -133,17 +133,18 @@ def setup_worker_connection(self, _worker_id):
         # reflected in django.db.connections. Otherwise new threads would
         # connect to the default database instead of the appropriate clone.
         start_method = multiprocessing.get_start_method()
-        if start_method == "fork":
+        alias = self.connection.alias
+        source_db_file = f"{alias}_{_worker_id}.sqlite3"
+        if start_method == "fork" or not os.path.isfile(source_db_file):
```

Comment:
:thinking: Should not we instead ask ourselves why this method is called when `other` db is not used?

Reply:
@shangxiao would you have any idea about the above?

Reply:
@felixxm I'm not sure I understand the question 🤔 (did you mean "when other _**is**_ used" instead?)  I was personally curious as to why this was needed for the 'other' alias - I thought I read somewhere that it was used to test db routing? which makes calling this pointless… hence why tests were passing with what appeared to be an empty db  (I've not been across what other is used for yet so correct me if i'm wrong)

To answer "why is this called when other _**is not**_ used" - it won't be called because there won't be an alias in the `DATABASES` setting 🤔

Reply:
Check out the ticket description, the `other` connection is not used by described subset of tests. As far as I'm aware that's why the clones of `other` databases are not cleaned up.

Reply:
Yes I did read the ticket description. The part that I mentioned I didn't understand was how the "other" alias was used in tests.

I understand how it works now though: I've read through the tests & the test runner and see how it selectively picks out "used" aliases from the test suite. It uses this list to then create & clone needed databases.

There's a gap between the test runner which knows about used aliases & creates/clones and `ParallelTestSuite` which is responsible for initialising the workers. Each worker simply initialises each alias in `django.db.connections` without caring about what is used & unused.

If you want it should be fairly simple to make worker initialisation aware of used databases: during `TestRunner._get_databases(suite)` you could pass the result into suite as well (if it's of type `ParallelTestSuite` that is).

Reply:
> If you want it should be fairly simple to make worker initialisation aware of used databases: during `TestRunner._get_databases(suite)` you could pass the result into suite as well (if it's of type `ParallelTestSuite` that is).

That would be great.